### DISCUSSION TITLE
How can I create a temporary file from an existing S3 file?

### DISCUSSION THREAD

**Discussion by lucasromanojf** [livewire/livewire#1089] [4. General]
Hello!

I'm trying to create a temporary file from an existing S3 file, but I am having hard time. Maybe someone could help me?

I have a file upload field inside an edit page. If a file already exists, I show a readonly filename with a delete button, where I can remove the current file and upload another one. But I should not delete the existing file until I hit the Save button.

So, I'm trying to create, in my Livewire component, a temporary file from my existing file, so I can delete it without problems, upload another one, and if I refresh the page, the existing file is there again, being changed only if I hit save.

Did anyone have this kind of need too? If so, did you solve this? How?

Thanks!

**Comment by devcircus** [comment]
Seems like there’s an easier way. I have the same setup where a user can replace an existing image and even revert back to the old image. You shouldn’t need a temporary file to handle this.

As long as you have the file name, you can delete the file whenever you want with:
```php
Storage::disk('s3')->delete('folder/filename.jpg');
```
It may help to have a flag variable that you can use to keep track of whether or not you want to show the old or the new. 
Is there some requirement that you generate the temp file?

**Reply by lucasromanojf** [reply]
I use the same form component for create and edit, so I am using the same public fields for create and update. It would be easier if I could just use the TemporaryUploadedFile for create and on submit acknowledge it using storeAs (already works fine this way on create), but also get the existing file from the model back to a TemporaryUploadedFile on edit, so it would not demand too many changes on the components. Also, on edit, using this approach, if I delete the file and refresh the page without saving, the existing file will be back there, untouched.
